web/app/routes/mainRoutes.js

```diff
@@ -0,0 +1,83 @@
+var express = require("express");
+var router = express.Router();
+var mongoose = require("mongoose");
+mongoose.connect("mongodb://localhost/routerdb");
+
+var userCtrl = require("../controllers/userCtrl");
+var drivingSchoolCtrl = require("../controllers/drivingSchoolCtrl");
+var drivingSessionCtrl = require("../controllers/drivingSessionCtrl");
+
+// server routes ===========================================================
+// handle things like api calls
+// authentication routes
+
+// Get the JSON for the student with the specified _id
+router.get("/students/:_id", function(req, res) {
+  userCtrl.getStudent(req, res);
```

Comment:
Long story short, I was saying you should keep the req and res objects at this layer instead of passing them down to a lower layer

I mean in general, this
```
router.post("/students/", function(req, res) {
    userCtrl.createStudent(req, res);
});
```
seems kinda useless, I mean, why even have this function if all it does is just pass the entire req and res down to another function.
It's functional the same as just doing
```
router.post("/students/", userCtrl.createStudent);
```


To simplify to what I said in my previous comment (I also just noticed that I have the app.get function calling the wrong lower level function, opps. Same idea still stands though), I think you should do the whole "separation of concerns" thing. Don't have the function that has to get the user data also have to format the response. The example I gave above was that you request the certain user's info, then also give 2 functions, one if the user's info gathering was successful, another if there was an error. That way, the lower level function doesn't need to know how to format the response or what the functions even do (this makes testing significantly easier fyi) to handle it's own job, which should be to purely get the user's information.